### PR TITLE
Removing duplicate model filtering, adding test coverage

### LINKED ISSUE #201: Search builder is filtering has_model twice
```
_query_:\"{!raw f=has_model_ssim}Image\" OR _query_:\"{!raw f=has_model_ssim}Video\" OR _query_:\"{!raw f=has_model_ssim}Collection\")", "has_model_ssim:(\"Image\" \"Video\" \"Collection\")"
````

### PR DESCRIPTION
**Pull request by escowles**
Fixes #201

### CODE REVIEW

File: app/controllers/concerns/curation_concerns/catalog_controller.rb [hunk 1]
```diff
@@ -14,7 +14,6 @@ module CurationConcerns::CatalogController
     # This applies appropriate access controls to all solr queries
     CatalogController.search_params_logic += [:add_access_controls_to_solr_params]
     self.search_params_logic += [:filter_models]
-    self.search_params_logic += [:only_curation_concerns]
```
Comment (cam156) [inline app/controllers/concerns/curation_concerns/catalog_controller.rb]: @escowles what is doing the duplicate filtering?  Seems strange to remove a line that was adderd by @jcoyne just a few days ago.

Comment (escowles) [inline app/controllers/concerns/curation_concerns/catalog_controller.rb]: filter_models and only_curation_concerns do basically the same thing -- and @jcoyne is the one who opened #201 and asked me to remove one of them.

Comment (jcoyne) [inline app/controllers/concerns/curation_concerns/catalog_controller.rb]: @cam156 I just changed the name of the method. The actual method was added by @flyingzumwalt See #201 for details of when that happened.

File: spec/search_builders/curation_concerns/search_builder_spec.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+    it 'does not filter results for admin users' do
+      expect(subject.gated_discovery_filters).to eq([])
+    end
+  end
+
+  describe '#discovery_permissions' do
+    before { allow(subject).to receive(:blacklight_params).and_return(works: 'mine') }
+
+    it 'limits query to edit permissions' do
```
Comment (jcoyne) [inline spec/search_builders/curation_concerns/search_builder_spec.rb]: This doesn't seem right.

Comment (jcoyne) [inline spec/search_builders/curation_concerns/search_builder_spec.rb]: I see. It's only when params[:works] = 'mine'.  Can we put this in a `context "when showing my works"` block?

**Comment by escowles** [comment]
@jcoyne: OK, I've added a context block (and also updated the filter_models test to use a regex so it's less brittle)

**Pull request merged**